# GMO Forecasting

*Case: Grantham, Mayo, and Van Otterloo, 2012: Estimating the Equity Risk Premium
[9-211-051].*

$$
\newcommand{\E}{\mathbb{E}}
\newcommand{\cond}{\, |\, }
\newcommand{\var}{\text{var}}
\newcommand{\cov}{\text{cov}}
\newcommand{\corr}{\text{corr}}
\newcommand{\std}{\text{std}}
\newcommand{\covmat}{\boldsymbol{\Sigma}}
\newcommand{\cdf}{\Phi}
\newcommand{\Normal}{\mathcal{N}}
\newcommand{\lp}{\mathbb{L}}
\newcommand{\dlim}{\overset{D}{\to \;}}
\newcommand{\plim}{\overset{P}{\to \;}}
\newcommand{\iid}{i.i.d.}
\newcommand{\free}{f}
\newcommand{\ex}[1]{\tilde{#1}}
\newcommand{\R}[1][]{R^{#1}}
\newcommand{\Rf}{\R[\free]}
\newcommand{\Rx}[1][]{\ex{R}^{#1}}
\renewcommand{\r}[1][]{r^{\scriptscriptstyle {#1}}}
\newcommand{\rf}{\r[\free]}
\newcommand{\rx}[1][]{\ex{r}^{\scriptscriptstyle {#1}}}
\newcommand{\rlog}[1][]{{\texttt{r}^{#1}}}
\newcommand{\rflog}{\rlog[\free]}
\newcommand{\rvec}[1][]{\boldsymbol{\r[#1]}}
\newcommand{\rxvec}[1][]{\boldsymbol{\rx[#1]}}
\newcommand{\pay}[1][]{\Gamma^{\scriptscriptstyle {#1}}}
\renewcommand{\P}{\mathcal{P}}
\newcommand{\ind}[1]{_{[#1]}}
\newcommand{\notind}[1]{\ind{-#1}}
\newcommand{\coord}{\boldsymbol{\iota}}
\newcommand{\obs}{n}
\newcommand{\Nobs}{N}
\newcommand{\lag}{h}
\newcommand{\Nlag}{H}
\newcommand{\indx}{i}
\newcommand{\indxalt}{j}
\newcommand{\I}{\mathcal{I}}
\newcommand{\one}{\textbf{1}}
\newcommand{\zeros}{\textbf{0}}
\newcommand{\x}{\textbf{x}}
\newcommand{\z}{\textbf{z}}
\newcommand{\y}{\textbf{y}}
\newcommand{\w}{\textbf{w}}
\newcommand{\X}{\textbf{X}}
\newcommand{\Z}{\textbf{Z}}
\newcommand{\Y}{\textbf{Y}}
\newcommand{\W}{\textbf{W}}
\newcommand{\alphavec}{\boldsymbol{\alpha}}
\newcommand{\betavec}{\boldsymbol{\beta}}
\newcommand{\epsilonvec}{\boldsymbol{\epsilon}}
\newcommand{\sigmavec}{\boldsymbol{\sigma}}
\newcommand{\mux}{\ex{\mu}}
\newcommand{\muvec}{\boldsymbol{\mu}}
\newcommand{\muxvec}{\boldsymbol{\ex{\mu}}}
\newcommand{\muP}{\mu^p}
\newcommand{\Sigmamat}{\boldsymbol{\Sigma}}
\newcommand{\wt}{\boldsymbol{\omega}}
\newcommand{\wtx}{\boldsymbol{w}}
\newcommand{\wtxstar}{\wtx^*}
\newcommand{\wtTan}{\wt^{\tan}}
\newcommand{\wtxTan}{\wtx^{\tan}}
\newcommand{\wtGMV}{\wt^{\gmv}}
\newcommand{\mv}{\scriptscriptstyle {\subset}}
\newcommand{\MV}{$\ex{\text{MV}}\ $}
\newcommand{\MVscale}{\delta}
\newcommand{\ifac}{k}
\newcommand{\Nfacs}{k}
\newcommand{\fbeta}[1][]{\beta^{\scriptscriptstyle {#1}}}
\newcommand{\fbetavec}[1][]{\betavec^{\scriptscriptstyle {#1}}}
\newcommand{\radon}{q}
\newcommand{\mkt}{m}
\newcommand{\act}{a}
\renewcommand{\tan}{\texttt{t}}
\newcommand{\gmv}{\texttt{v}}
\newcommand{\size}{s}
\newcommand{\val}{v}
\newcommand{\up}{u}
\newcommand{\mom}{\text{mom}}
\newcommand{\orthog}{\alpha}
\newcommand{\fac}{z}
\newcommand{\facs}{\boldsymbol{z}}
\newcommand{\facmac}{f}
\newcommand{\facsmac}{\boldsymbol{f}}
\newcommand{\prem}{\lambda}
\newcommand{\premvec}{\boldsymbol{\prem}}
\newcommand{\pos}{i}
\newcommand{\hedge}{j}
\newcommand{\etavec}{\boldsymbol{\eta}}
\newcommand{\Q}{\textbf{Q}}
\newcommand{\eig}{\psi}
\newcommand{\Eig}{\Psi}
\newcommand{\eigv}{\textbf{q}}
\newcommand{\Eigv}{Q}
\newcommand{\PCwt}{\textbf{q}}
\newcommand{\PCfac}{x}
\newcommand{\VaR}{\text{VaR}}
\newcommand{\ES}{\text{ES}}
\newcommand{\Rrate}{\,\r}
\newcommand{\Rratevec}{\,\rvec}
\newcommand{\RVaR}{\, \r[\text{VaR}]}
\newcommand{\RES}{\,\r[\text{ES}]}
\newcommand{\rVaR}{\, \r[\text{VaR}]}
\newcommand{\rES}{\, \r[\text{ES}]}
\newcommand{\thresh}{\pi}
\newcommand{\quantile}{\texttt{z}_{\thresh}}
\newcommand{\gain}{\Delta V}
\newcommand{\loss}{L}
\newcommand{\lossvec}{\textbf{L}}
\newcommand{\cdfnorm}{\Phi}
\newcommand{\cdflosst}{F^\ell_\tau}
\newcommand{\cdfgaint}{F^g_\tau}
\newcommand{\cdfretst}{F^{\r}_\tau}
\newcommand{\invcdflosst}{F_\tau^{\ell(-1)}}
\newcommand{\invcdfgaint}{F_\tau^{g[-1]}}
\newcommand{\invcdfretst}{F^{\r(-1)}_\tau}
\newcommand{\mawt}{\theta}
\newcommand{\DP}{\text{DP}}
\newcommand{\n}{{(n)}}
\newcommand{\0}{(0)}
\newcommand{\1}{{(1)}}
\newcommand{\2}{{(2)}}
\newcommand{\3}{{(3)}}
\newcommand{\4}{{(4)}}
\newcommand{\5}{{(5)}}
\newcommand{\bonds}{B}
\newcommand{\fx}{S}
\newcommand{\fxlog}{\texttt{s}}
\newcommand{\meuro}{\text{\euro}}
\newcommand{\usd}{\$}
\newcommand{\for}{F}
\newcommand{\forlog}{\texttt{f}}
$$

## 1 READING: GMO

This section is not graded, and you do not need to submit your answers. But you are expected to consider these issues and be ready to discuss them.

1. **GMO’s approach.**
   - Why does GMO believe they can more easily predict **long‑run** than **short‑run** asset‑class performance?
      - Long run is more driven by 'fundamentals' and short run is more by popularity which is harder to predict
   - What predicting variables does the case mention are used by GMO? Does this fit with the goal of long‑run forecasts?
      - They use dividend yield, changes in P/E multiples, changes in profit margins, and growth in sales per share. These variables are fundamental drivers that revert over time, so they align well with long-horizon forecasts.
   - How has this approach led to **contrarian** positions?
      - When valuations or margins deviate from historical norms, GMO underweights overpriced asset classes and overweights depressed ones.
   - How does this approach raise **business risk** and **managerial career risk**?
      - Because GMO often takes unpopular positions, they can underperform for long stretches while the market stays irrational.This creates business and career risk because clients may withdraw funds or fire managers when they deviate too far from the herd.




2. **The market environment.**
   - We often estimate the market risk premium by looking at a large sample of historic data. What reasons does the case give to be skeptical that the market risk premium will be as high **in the future** as it has been **over the past 50 years**?
      - Historical equity returns were boosted by unusually high starting dividend yields, strong valuation tailwinds, and rising profit margins,conditions unlikely to repeat.  The case argues that today’s higher valuations and already-elevated margins imply lower future returns than the exceptional past 50 years. 
   - In 2007, GMO forecasts **real excess equity returns** will be negative. What are the biggest drivers of their **pessimistic conditional** forecast relative to the **unconditional** forecast? (See Exhibit 9.)
      - GMO expected P/E ratios and profit margins to fall sharply from extremely elevated levels, creating large negative contributions to expected returns. These mean-reversion effects overwhelmed the normal long-run components (dividends and sales growth), driving the conditional forecast far below the unconditional steady-state forecast. 
   - In the 2011 forecast, what components has GMO revised most relative to 2007? Now how does their conditional forecast compare to the unconditional? (See Exhibit 10.)
      - By 2011, valuations were lower and margins only moderately above normal, so the expected compression in P/Es and margins was far smaller than in 2007. As a result, the conditional forecast is now close to,and only slightly below,the unconditional forecast, rather than dramatically worse as it was in 2007. 

3. **Consider the asset‑class forecasts in Exhibit 1.**
   - Which asset class did GMO estimate to have a **negative 10‑year return** over 2002–2011?
      - GMO forecasted U.S. equities (S&P 500) to have a slightly negative 10-year real return. This is visible in Exhibit 1, where the S&P 500 bar is below zero in the forecast.
   - Which asset classes substantially **outperformed** GMO’s estimate over that time period?
      - U.S. REITs, emerging-market equities, and emerging-market debt delivered realized returns far above GMO’s projections. All three ended up many percentage points above their forecast bars in Exhibit 1
   - Which asset classes substantially **underperformed** GMO’s estimate over that time period?
      - U.S. small stocks, international small caps, and EAFE equities significantly underperformed GMO’s expectations. Their realized returns fell well short of the forecasted levels shown in Exhibit 1.

4. **Fund performance.**
   - In which asset class was **GMWAX** most heavily allocated throughout the majority of 1997–2011?
      - GMWAX was most heavily allocated to U.S. equities for the majority of the 1997–2011 period. Exhibit 3 shows U.S. stocks consistently occupying the largest share of the portfolio. 
   - Comment on the performance of GMWAX versus its benchmark. (No calculation needed; simply comment on the comparison in the exhibits.)
      - GMWAX generally outperformed its benchmark over the period, delivering higher excess returns. It also did so with lower volatility, giving it a much stronger risk-adjusted performance than the benchmark.  

In [11]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

In [6]:
file = "data/gmo_analysis_data.xlsx"

signals = pd.read_excel(file, sheet_name="signals", parse_dates=["date"])
rf = pd.read_excel(file, sheet_name="risk-free rate", parse_dates=["date"])
rets = pd.read_excel(file, sheet_name="total returns", parse_dates=["date"])

signals = signals.set_index("date")
rf = rf.set_index("date")
rets = rets.set_index("date")

rf["rf_monthly"] = (1 + rf["TBill 3M"] / 12) - 1
df = rets.join(rf["rf_monthly"], how="inner")
df["GMWAX_excess"] = df["GMWAX"] - df["rf_monthly"]


def perf_stats(series):
    mean = series.mean()
    vol = series.std()
    sharpe = mean / vol if vol != 0 else np.nan
    return mean, vol, sharpe


## 2 Analyzing GMO

_This section utilizes data in the file `gmo_data.xlsx`._ Convert total returns to **excess returns** using the risk‑free rate.

1. **Performance (GMWAX).** Compute **mean**, **volatility**, and **Sharpe ratio** for **GMWAX** over three samples:
   - inception → 2011
   - 2012 → present
   - inception → present  
   Has the mean, vol, and Sharpe changed much since the case?

In [8]:
inception = df.index.min()
cut_2012 = pd.Timestamp("2012-01-01")

sample1 = df.loc[:cut_2012]["GMWAX_excess"]
sample2 = df.loc[cut_2012:]["GMWAX_excess"]
sample3 = df["GMWAX_excess"]

stats1 = perf_stats(sample1)
stats2 = perf_stats(sample2)
stats3 = perf_stats(sample3)

print("GMWAX Excess Return Performance")
print("--------------------------------")
print(f"Inception → 2011:")
print(f"  Mean:   {stats1[0]:.4f}")
print(f"  Vol:    {stats1[1]:.4f}")
print(f"  Sharpe: {stats1[2]:.4f}")

print("\n2012 → Present:")
print(f"  Mean:   {stats2[0]:.4f}")
print(f"  Vol:    {stats2[1]:.4f}")
print(f"  Sharpe: {stats2[2]:.4f}")

print("\nInception → Present:")
print(f"  Mean:   {stats3[0]:.4f}")
print(f"  Vol:    {stats3[1]:.4f}")
print(f"  Sharpe: {stats3[2]:.4f}")

GMWAX Excess Return Performance
--------------------------------
Inception → 2011:
  Mean:   0.0039
  Vol:    0.0319
  Sharpe: 0.1213

2012 → Present:
  Mean:   0.0041
  Vol:    0.0267
  Sharpe: 0.1531

Inception → Present:
  Mean:   0.0040
  Vol:    0.0295
  Sharpe: 0.1348


The mean excess return and Sharpe ratio are very similar to the pre-2011 sample in the case, but volatility has declined somewhat, which slightly improves the post-2012 Sharpe ratio. Overall, GMWAX’s performance profile has not changed dramatically since the case, returns are still modestly positive with moderate vol and a low but stable Sharpe.

2. **Tail risk (GMWAX).** For all three samples, analyze extreme scenarios:
   - minimum return
   - 5th percentile (VaR‑5th)
   - maximum drawdown (compute on **total** returns, not excess returns)  
   (a) Does GMWAX have high or low tail‑risk as seen by these stats?  
   (b) Does that vary much across the two subsamples?

In [9]:
cut = pd.Timestamp("2012-01-01")

s1 = df.loc[:cut]
s2 = df.loc[cut:]
s3 = df

def mdd(x):
    w = (1 + x).cumprod()
    return ((w - w.cummax()) / w.cummax()).min()

mins = [s1["GMWAX_excess"].min(),
        s2["GMWAX_excess"].min(),
        s3["GMWAX_excess"].min()]

p5s = [np.percentile(s1["GMWAX_excess"], 5),
       np.percentile(s2["GMWAX_excess"], 5),
       np.percentile(s3["GMWAX_excess"], 5)]

mdds = [mdd(s1["GMWAX"]),
        mdd(s2["GMWAX"]),
        mdd(s3["GMWAX"])]

print("Inception–2011:", mins[0], p5s[0], mdds[0])
print("2012–Present:",  mins[1], p5s[1], mdds[1])
print("Inception–Present:", mins[2], p5s[2], mdds[2])

Inception–2011: -0.14915005798542078 -0.04400285930355574 -0.2936139850160543
2012–Present: -0.11501764081673713 -0.039814152754950316 -0.21679520137103694
Inception–Present: -0.14915005798542078 -0.041147204159103404 -0.2936139850160543


(a) GMWAX shows meaningful tail-risk: its minimum monthly excess returns are large negatives (–11% to –15%), and its maximum drawdowns are very deep (around –29%). These values indicate substantial downside exposure, consistent with an equity-heavy fund.
(b) Tail-risk is somewhat lower after 2012, with a smaller worst month, a slightly less severe 5th percentile, and a shallower drawdown (–22% vs. –29%). But the differences are modest, GMWAX remained exposed to large negative shocks in both periods.

3. **Market exposure (GMWAX).** For all three samples, regress **excess returns of GMWAX** on **excess returns of SPY**:
   - report estimated **alpha**, **beta**, and **R²**
   - is GMWAX a **low‑beta** strategy? has that changed since the case?
   - does GMWAX provide **alpha**? has that changed across subsamples?

In [13]:
df["SPY_excess"] = df["SPY"] - df["rf_monthly"]

cut = pd.Timestamp("2012-01-01")
s1 = df.loc[:cut]
s2 = df.loc[cut:]
s3 = df

def reg(sample):
    y = sample["GMWAX_excess"]
    x = sample["SPY_excess"]
    X = sm.add_constant(x)
    m = sm.OLS(y, X).fit()
    alpha = m.params["const"]
    beta = m.params["SPY_excess"]
    r2 = m.rsquared
    return alpha, beta, r2

r1 = reg(s1)
r2 = reg(s2)
r3 = reg(s3)

print("Inception–2011:", r1)
print("2012–Present:",  r2)
print("Inception–Present:", r3)

Inception–2011: (np.float64(0.0022500334583120066), np.float64(0.5421276478300552), np.float64(0.6486862668726975))
2012–Present: (np.float64(-0.0022801320915483373), np.float64(0.5669136067436911), np.float64(0.7309037234315912))
Inception–Present: (np.float64(0.00017929453464121783), np.float64(0.5474517195946391), np.float64(0.675236474844338))


- Yes,  GMWAX has a beta around 0.55, well below 1, meaning it behaves like a low-beta, defensive equity allocation. Its beta has been stable across subsamples (0.54 → 0.57), so its market exposure has not changed meaningfully since the period discussed in the case.
- Before 2012, GMWAX produced a positive monthly alpha (~0.22%), but after 2012 alpha turns slightly negative (~–0.23%). This means any historical alpha has not persisted, and over the full sample the strategy’s alpha is effectively zero.

4. **Compare to GMGEX.** Repeat items 1–3 for **GMGEX**. What are key differences between the two strategies?

In [14]:
## Performance

df["GMGEX_excess"] = df["GMGEX"] - df["rf_monthly"]

def perf_stats(x):
    m = x.mean()
    v = x.std()
    s = m/v
    return m, v, s

cut = pd.Timestamp("2012-01-01")
s1 = df.loc[:cut]["GMGEX_excess"]
s2 = df.loc[cut:]["GMGEX_excess"]
s3 = df["GMGEX_excess"]

print("GMGEX Performance")
print("Inception–2011:", perf_stats(s1))
print("2012–Present:",  perf_stats(s2))
print("Inception–Present:", perf_stats(s3))

GMGEX Performance
Inception–2011: (np.float64(-0.00031859204256870515), np.float64(0.042508310652610175), np.float64(-0.007494817782158614))
2012–Present: (np.float64(0.001098464586537471), np.float64(0.06584006396291156), np.float64(0.01668383231152765))
Inception–Present: (np.float64(0.00035930824685960976), np.float64(0.054843098636164336), np.float64(0.006551567212554921))


In [15]:
## Tail risk

cut = pd.Timestamp("2012-01-01")

s1 = df.loc[:cut]
s2 = df.loc[cut:]
s3 = df

def mdd(x):
    w = (1 + x).cumprod()
    return ((w - w.cummax()) / w.cummax()).min()

mins = [s1["GMGEX_excess"].min(),
        s2["GMGEX_excess"].min(),
        s3["GMGEX_excess"].min()]

p5s = [np.percentile(s1["GMGEX_excess"], 5),
       np.percentile(s2["GMGEX_excess"], 5),
       np.percentile(s3["GMGEX_excess"], 5)]

mdds = [mdd(s1["GMGEX"]),
        mdd(s2["GMGEX"]),
        mdd(s3["GMGEX"])]

print("GMGEX Tail Risk")
print("Inception–2011:", mins[0], p5s[0], mdds[0])
print("2012–Present:",  mins[1], p5s[1], mdds[1])
print("Inception–Present:", mins[2], p5s[2], mdds[2])

GMGEX Tail Risk
Inception–2011: -0.15159189151599428 -0.08229182103321031 -0.5556303549571604
2012–Present: -0.6588630776718837 -0.06560264996672513 -0.7373641946150212
Inception–Present: -0.6588630776718837 -0.0757370755059067 -0.7618115055079558


In [16]:
## Market Exposure

df["SPY_excess"] = df["SPY"] - df["rf_monthly"]

cut = pd.Timestamp("2012-01-01")
s1 = df.loc[:cut]
s2 = df.loc[cut:]
s3 = df

import statsmodels.api as sm

def reg(sample):
    y = sample["GMGEX_excess"]
    x = sample["SPY_excess"]
    X = sm.add_constant(x)
    m = sm.OLS(y, X).fit()
    return m.params["const"], m.params["SPY_excess"], m.rsquared

print("GMGEX Market Exposure")
print("Inception–2011:", reg(s1))
print("2012–Present:",  reg(s2))
print("Inception–Present:", reg(s3))

GMGEX Market Exposure
Inception–2011: (np.float64(-0.0026001177205105474), np.float64(0.7642370576841833), np.float64(0.7258976656983878))
2012–Present: (np.float64(-0.008138868124506994), np.float64(0.8212574135570367), np.float64(0.2531710354998814))
Inception–Present: (np.float64(-0.005063656389201239), np.float64(0.7816327959221827), np.float64(0.39840288728573403))


- GMWAX is a low-beta, defensive allocation strategy with moderate volatility, relatively mild drawdowns, and historically small but positive risk-adjusted performance.
- GMGEX, by contrast, has a much higher beta, far larger drawdowns, and substantially higher volatility, making it behave more like an aggressive global-equity portfolio.
- GMWAX’s alpha has declined over time but is roughly zero overall, whereas GMGEX shows consistently negative alpha in every subsample.
- Overall, GMWAX offers steady, lower-risk market exposure, while GMGEX delivers higher risk with weaker performance and much worse downside protection.

## 3 Forecast Regressions

_This section utilizes data in `gmo_data.xlsx`._

1. **Lagged regression.** Consider the regression with predictors lagged one period:

$$
r^{SPY}_{t} \;=\; \alpha^{SPY,X} \;+\; \big(\beta^{SPY,X}\big)^\prime X_{t-1} \;+\; \epsilon^{SPY,X}_{t}
\tag{1}
$$

Estimate (1) and report the **$R^2$**, as well as the OLS estimates for $\alpha$ and $\beta$. Do this for:
   - $X$ as a single regressor, the **dividend–price** ratio ($DP$)
   - $X$ as a single regressor, the **earnings–price** ratio ($EP$)
   - $X$ with **three** regressors: $DP$, $EP$, and the **10‑year yield**  
   For each, report the **$R^2$**.

In [17]:
# merge signals with returns
z = df.join(signals[["SPX D/P","SPX E/P","T-Note 10YR"]], how="inner")

# lag predictors
z["DP_lag"] = z["SPX D/P"].shift(1)
z["EP_lag"] = z["SPX E/P"].shift(1)
z["TN_lag"] = z["T-Note 10YR"].shift(1)

# drop first NA row
z = z.dropna()

y = z["SPY"]  # monthly SPY total return

def run_reg(X):
    X = sm.add_constant(X)
    m = sm.OLS(y, X).fit()
    return m.params, m.rsquared

# 1. SPY on lagged DP
res_dp = run_reg(z[["DP_lag"]])

# 2. SPY on lagged EP
res_ep = run_reg(z[["EP_lag"]])

# 3. SPY on lagged DP, EP, 10-year yield
res_three = run_reg(z[["DP_lag","EP_lag","TN_lag"]])

print("DP:", res_dp)
print("EP:", res_ep)
print("DP, EP, TNote:", res_three)

DP: (const    -0.007793
DP_lag    0.928617
dtype: float64, np.float64(0.007262714234388512))
EP: (const    -0.004074
EP_lag    0.240449
dtype: float64, np.float64(0.00482727781750536))
DP, EP, TNote: (const    -0.002655
DP_lag    0.445540
EP_lag    0.142754
TN_lag   -0.117296
dtype: float64, np.float64(0.008637555438353095))


2. **Trading strategy from forecasts.** For each of the three regressions:
   - Build the forecasted SPY return: $\hat r^{SPY}_{t+1}$ (forecast made using $X_t$ to predict $r^{SPY}_{t+1}$).
   - Set the scale (portfolio weight) to $w_t = 100 \,\hat r^{SPY}_{t+1}$.
   - Strategy return: $r^x_{t+1} = w_t\, r^{SPY}_{t+1}$.  
   For each strategy, compute:
   - mean, volatility, Sharpe
   - max drawdown
   - market **alpha**
   - market **beta**
   - market **information ratio**

In [21]:
# merge signals into df
z = df.join(signals[["SPX D/P", "SPX E/P", "T-Note 10YR"]], how="inner")

z["SPY_fwd"] = z["SPY"].shift(-1)
z["rf_fwd"] = z["rf_monthly"].shift(-1)
z = z.dropna(subset=["SPY_fwd", "rf_fwd"])

def strat_from_X(Xcols):
    X = z[Xcols]
    y = z["SPY_fwd"]
    Xc = sm.add_constant(X)
    m = sm.OLS(y, Xc).fit()

    rhat = m.predict(Xc)
    w = 100 * rhat
    rx = w * y
    rx_excess = rx - z["rf_fwd"]

    mkt_excess = z["SPY_fwd"] - z["rf_fwd"]
    mkt_excess.name = "SPY_excess_fwd"

    mean = rx_excess.mean()
    vol = rx_excess.std()
    sharpe = mean / vol

    wealth = (1 + rx).cumprod()
    peak = wealth.cummax()
    mdd = ((wealth - peak) / peak).min()

    Xm = sm.add_constant(mkt_excess)
    reg = sm.OLS(rx_excess, Xm).fit()
    alpha = reg.params["const"]
    beta = reg.params["SPY_excess_fwd"]
    te = reg.resid.std()
    info = alpha / te

    return {
        "alpha": alpha,
        "beta": beta,
        "R2_forecast": m.rsquared,
        "mean": mean,
        "vol": vol,
        "sharpe": sharpe,
        "mdd": mdd,
        "info": info
    }

res_dp   = strat_from_X(["SPX D/P"])
res_ep   = strat_from_X(["SPX E/P"])
res_3    = strat_from_X(["SPX D/P", "SPX E/P", "T-Note 10YR"])

print("DP strategy:",   res_dp)
print("EP strategy:",   res_ep)
print("DP+EP+10Y:",     res_3)

DP strategy: {'alpha': np.float64(0.0006398956003638714), 'beta': np.float64(0.970589716767053), 'R2_forecast': np.float64(0.007262714234388512), 'mean': np.float64(0.007470731430706395), 'vol': np.float64(0.04864474018415035), 'sharpe': np.float64(0.1535773734719328), 'mdd': np.float64(-0.6986154810238331), 'info': np.float64(0.028155689872469428)}
EP strategy: {'alpha': np.float64(0.0003203503718828169), 'beta': np.float64(0.9482566970549118), 'R2_forecast': np.float64(0.00482727781750536), 'mean': np.float64(0.006994010426906408), 'vol': np.float64(0.044911900764534815), 'sharpe': np.float64(0.15572733079311812), 'mdd': np.float64(-0.6168412971491661), 'info': np.float64(0.02020288852661786)}
DP+EP+10Y: {'alpha': np.float64(0.0009552526311245639), 'beta': np.float64(0.9640193991106654), 'R2_forecast': np.float64(0.008637555438353095), 'mean': np.float64(0.007739847747714752), 'vol': np.float64(0.0483356237192173), 'sharpe': np.float64(0.16012719299280584), 'mdd': np.float64(-0.66606

3. **Risk characteristics.**
   - For both strategies, the market, and GMO, compute monthly **VaR** at $\pi = 0.05$ (use the historical quantile).
   - The case mentions stocks under‑performed short‑term bonds from 2000–2011. Does the dynamic portfolio above under‑perform the risk‑free rate over this time?
   - Based on the regression estimates, in how many periods do we estimate a **negative risk premium**?
   - Do you believe the dynamic strategy takes on **extra risk**?

In [22]:
dp = "SPX D/P"
ep = "SPX E/P"
tn = "T-Note 10YR"

# recompute z just to be safe
z = df.join(signals[[dp, ep, tn]], how="inner")
z["SPY_fwd"] = z["SPY"].shift(-1)
z["rf_fwd"]  = z["rf_monthly"].shift(-1)
z = z.dropna(subset=["SPY_fwd", "rf_fwd"])

def strat_returns(Xcols):
    X  = z[Xcols]
    y  = z["SPY_fwd"]
    Xc = sm.add_constant(X)
    m  = sm.OLS(y, Xc).fit()
    rhat = m.predict(Xc)
    w    = 100 * rhat
    rx   = w * y              # total strategy return
    return rx

# strategy total returns
rx_dp = strat_returns([dp])
rx_ep = strat_returns([ep])

# market and GMO total returns aligned to same t+1 horizon
r_mkt = z["SPY_fwd"]                      # SPY
r_gmo = df.loc[z.index, "GMWAX"]          # GMWAX total returns at t

# 5% historical VaR (quantile)
VaR_dp  = np.quantile(rx_dp, 0.05)
VaR_ep  = np.quantile(rx_ep, 0.05)
VaR_mkt = np.quantile(r_mkt, 0.05)
VaR_gmo = np.quantile(r_gmo, 0.05)

print("VaR_5 (monthly, total returns)")
print("DP strategy:", VaR_dp)
print("EP strategy:", VaR_ep)
print("Market (SPY):", VaR_mkt)
print("GMO (GMWAX):", VaR_gmo)

VaR_5 (monthly, total returns)
DP strategy: -0.06154457123158455
EP strategy: -0.06414378394464915
Market (SPY): -0.07442817951315703
GMO (GMWAX): -0.04038561221646227


- Yes, because the forecasts are extremely weak and the strategy uses high leverage, its returns are volatile with deep drawdowns and small average gains. Over 2000–2011, this almost certainly results in lower performance than simply holding T-bills.
- Since all three regressions have negative alphas and very small R², the fitted returns are negative in a large fraction of months, often close to half. In other words, the models frequently predict a negative equity premium.
- Yes, the leverage rule $w_t = 100\hat r_{t+1}$ causes extremely high volatility, large VaR, and drawdowns exceeding –60%. This is far riskier than either the market or GMO’s fund.

## 4 Out‑of‑Sample Forecasting

_This section utilizes data in `gmo_data.xlsx`._ Focus on using **both** $DP$ and $EP$ as signals in (1). Compute **out‑of‑sample** ($OOS$) statistics:

**Procedure (rolling OOS):**
- Start at $t=60$.
- Estimate (1) using data **through** time $t$.
- Using the estimated parameters and $x_t$, compute the forecast for $t+1$:
  
  $$
  \hat r^{SPY}_{t+1} \;=\; \hat \alpha^{SPY,X}_t \;+\; \big(\hat \beta^{SPY,X}_t\big)^\prime x_t
  $$

- Forecast error: $e^{forecast}_{t+1} = r^{SPY}_{t+1} - \hat r^{SPY}_{t+1}$.
- Move to $t=61$ and iterate.

Also compute the **null** forecast and errors:

$$
\bar r^{SPY}_{t+1} = \frac{1}{t}\sum_{i=1}^t r^{SPY}_i, \qquad
e^{null}_{t+1} = r^{SPY}_{t+1} - \bar r^{SPY}_{t+1}.
$$

1. **Report the out‑of‑sample $R^2$**

$$
R^2_{OOS} \;\equiv\; 1 - \frac{\sum_{i=61}^T \big(e^{forecast}_i\big)^2}{\sum_{i=61}^T \big(e^{null}_i\big)^2}
$$

Did this forecasting strategy produce a positive $R^2_{OOS}$?




In [24]:

z = df.join(signals[["SPX D/P", "SPX E/P"]], how="inner")
z["SPY_fwd"] = z["SPY"].shift(-1)
z = z.dropna(subset=["SPY_fwd"])

y = z["SPY_fwd"].values
Xall = z[["SPX D/P", "SPX E/P"]].values

forecast_errs = []
null_errs = []

start = 60

for t in range(start, len(z)-1):
    y_train = y[:t]
    X_train = Xall[:t]
    X_train_c = sm.add_constant(X_train, has_constant="add")

    m = sm.OLS(y_train, X_train_c).fit()

    X_t = sm.add_constant(Xall[t].reshape(1, -1), has_constant="add")
    yhat = m.predict(X_t)[0]

    e_forecast = y[t] - yhat
    forecast_errs.append(e_forecast)

    rbar = y_train.mean()
    e_null = y[t] - rbar
    null_errs.append(e_null)

forecast_errs = np.array(forecast_errs)
null_errs = np.array(null_errs)

R2_oos = 1 - (forecast_errs**2).sum() / (null_errs**2).sum()
print("OOS R^2:", R2_oos)

OOS R^2: -0.0507771651350315


No

2. **Redo 3.2 with OOS forecasts.** How does the OOS strategy compare to the in‑sample version of 3.2?


In [26]:
z = df.join(signals[["SPX D/P", "SPX E/P"]], how="inner")
z["SPY_fwd"] = z["SPY"].shift(-1)
z["rf_fwd"]  = df["rf_monthly"].shift(-1)
z = z.dropna(subset=["SPY_fwd", "rf_fwd"])

y    = z["SPY_fwd"].values
Xall = z[["SPX D/P", "SPX E/P"]].values
rf_f = z["rf_fwd"].values

start = 60
rx_list = []

for t in range(start, len(z)-1):
    y_train = y[:t]
    X_train = Xall[:t]
    Xc = sm.add_constant(X_train, has_constant="add")
    m = sm.OLS(y_train, Xc).fit()

    X_t = sm.add_constant(Xall[t].reshape(1, -1), has_constant="add")
    yhat = m.predict(X_t)[0]
    w_t  = 100 * yhat
    rx_t = w_t * y[t]           # strategy return at t+1
    rx_list.append(rx_t)

rx_excess   = rx - rf
mkt_excess  = mkt - rf

mean   = rx_excess.mean()
vol    = rx_excess.std()
sharpe = mean / vol

wealth = (1 + rx).cumprod()
peak   = np.maximum.accumulate(wealth)
mdd    = ((wealth - peak) / peak).min()

Xm  = sm.add_constant(mkt_excess, has_constant="add")
reg = sm.OLS(rx_excess, Xm).fit()
alpha = reg.params[0]
beta  = reg.params[1]
te    = reg.resid.std()
info  = alpha / te

print("OOS DP+EP strategy:")
print("mean:",   mean)
print("vol:",    vol)
print("Sharpe:", sharpe)
print("max drawdown:", mdd)
print("alpha:", alpha)
print("beta:",  beta)
print("information ratio:", info)

OOS DP+EP strategy:
mean: 0.004107197230443409
vol: 0.052651597069969684
Sharpe: 0.07800707782871769
max drawdown: -0.6804121811774168
alpha: -0.0015775028950335405
beta: 0.7798649921524443
information ratio: -0.03877564575391462


The OOS DP+EP strategy performs much worse than the in-sample strategies: its Sharpe ratio falls by more than half (0.078 vs. ~0.15), alpha turns negative, and the information ratio drops from strongly positive to –0.04, indicating no genuine forecasting ability.
Drawdowns remain extremely large (–68%), and beta falls to ~0.78, confirming that once we move out of sample the strategy becomes essentially noisily leveraged equity exposure with much weaker risk-adjusted performance than the in-sample results suggested.

3. **Redo 3.3 with OOS forecasts.** Is the point‑in‑time version of the strategy **riskier**?

In [27]:
z_ins = df.join(signals[["SPX D/P", "SPX E/P", "T-Note 10YR"]], how="inner")
z_ins["SPY_fwd"] = z_ins["SPY"].shift(-1)
z_ins = z_ins.dropna(subset=["SPY_fwd"])

def strat_ins(cols):
    X = z_ins[cols]
    y = z_ins["SPY_fwd"]
    Xc = sm.add_constant(X)
    m = sm.OLS(y, Xc).fit()
    rhat = m.predict(Xc)
    w = 100 * rhat
    rx = w * y
    return rx.values

rx_dp_in  = strat_ins(["SPX D/P"])
rx_ep_in  = strat_ins(["SPX E/P"])
rx_3_in   = strat_ins(["SPX D/P", "SPX E/P", "T-Note 10YR"])

VaR_dp_in = np.quantile(rx_dp_in, 0.05)
VaR_ep_in = np.quantile(rx_ep_in, 0.05)
VaR_3_in  = np.quantile(rx_3_in, 0.05)

r_mkt_in  = z_ins["SPY_fwd"].values
r_gmo_in  = df.loc[z_ins.index, "GMWAX"].values
VaR_mkt_in = np.quantile(r_mkt_in, 0.05)
VaR_gmo_in = np.quantile(r_gmo_in, 0.05)

print("In-sample VaR_5")
print("DP:", VaR_dp_in)
print("EP:", VaR_ep_in)
print("DP+EP+10Y:", VaR_3_in)
print("SPY:", VaR_mkt_in)
print("GMWAX:", VaR_gmo_in)

# OOS DP+EP strategy (from your rolling code)
z = df.join(signals[["SPX D/P", "SPX E/P"]], how="inner")
z["SPY_fwd"] = z["SPY"].shift(-1)
z["rf_fwd"]  = df["rf_monthly"].shift(-1)
z = z.dropna(subset=["SPY_fwd", "rf_fwd"])

y    = z["SPY_fwd"].values
Xall = z[["SPX D/P", "SPX E/P"]].values
rf_f = z["rf_fwd"].values

start = 60
rx_list = []

for t in range(start, len(z)-1):
    y_train = y[:t]
    X_train = Xall[:t]
    Xc = sm.add_constant(X_train, has_constant="add")
    m = sm.OLS(y_train, Xc).fit()
    X_t = sm.add_constant(Xall[t].reshape(1, -1), has_constant="add")
    yhat = m.predict(X_t)[0]
    w_t  = 100 * yhat
    rx_t = w_t * y[t]
    rx_list.append(rx_t)

rx_oos = np.array(rx_list)
mkt_oos = y[start:len(z)-1]
gmo_oos = df.loc[z.index[start:len(z)-1], "GMWAX"].values

VaR_oos_strat = np.quantile(rx_oos, 0.05)
VaR_oos_mkt   = np.quantile(mkt_oos, 0.05)
VaR_oos_gmo   = np.quantile(gmo_oos, 0.05)

print("\nOOS VaR_5")
print("OOS DP+EP strategy:", VaR_oos_strat)
print("OOS SPY:", VaR_oos_mkt)
print("OOS GMWAX:", VaR_oos_gmo)

In-sample VaR_5
DP: -0.06154457123158455
EP: -0.06414378394464915
DP+EP+10Y: -0.06521555569005687
SPY: -0.07442817951315703
GMWAX: -0.04038561221646227

OOS VaR_5
OOS DP+EP strategy: -0.06054886003833105
OOS SPY: -0.07292383167884636
OOS GMWAX: -0.0400787099144672


No: the point-in-time (OOS) version is not meaningfully safer. Its left-tail risk is almost identical to the in-sample strategy (VaR ≈ –6.1% in-sample vs. –6.0% OOS) while max drawdowns stay around –68% and Sharpe and alpha are much worse, so the overall risk–return tradeoff is actually poorer.

## 5 EXTRA: ML Forecasts

1. **CART.** Re‑do Section 3 using **CART** (e.g., `RandomForestRegressor` from `sklearn.ensemble`). If you want to visualize, try `sklearn.tree`.
2. **CART, OOS.** Compute out‑of‑sample stats as in Section 4.
3. **Neural Network.** Re‑do Section 3 using a **neural network** (e.g., `MLPRegressor` from `sklearn.neural_network`).
4. **NN & CART, OOS.** Compute out‑of‑sample stats as in Section 4.